plugins/radar-android-google-sleep/src/main/java/org.radarbase.passive.google.sleep/SleepReceiver.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.radarbase.passive.google.sleep
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.google.android.gms.location.SleepClassifyEvent
+import com.google.android.gms.location.SleepSegmentEvent
+
+class SleepReceiver(private val googleSleepManager: GoogleSleepManager) : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent?) {
+            intent ?: return
+            when {
+                SleepSegmentEvent.hasEvents(intent) -> googleSleepManager.sendSleepSegmentData(intent)
+                SleepClassifyEvent.hasEvents(intent) -> googleSleepManager.sendSleepClassifyData(intent)
+            }
```

Comment:
I'm not sure if an intent can have both segment and classify events. If so, this should be two independent if statements
```suggestion
            if (SleepSegmentEvent.hasEvents(intent)) googleSleepManager.sendSleepSegmentData(intent)
            if (SleepClassifyEvent.hasEvents(intent)) -> googleSleepManager.sendSleepClassifyData(intent)
```